core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -339,6 +344,9 @@ protected <T> void internalSet(int pos, T value) {
         this.sortOrderId = (Integer) value;
         return;
       case 17:
+        this.referencedDataFile = value != null ? value.toString() : null;
+        return;
+      case 18:
```

Comment:
This changes the ordinal of `fileOrdinal`. This normally doesn't matter as we read with a projection. We also changed it a couple of times before. Anything I missed?

Reply:
As long as both reads and writes are updated, it should be fine.

Reply:
Thanks for confirming!